helpers/aave/buildGetTokenFunction.ts

```diff
@@ -1,29 +1,48 @@
-import { AAVETokens, ADDRESSES, ONE } from '@oasisdex/oasis-actions/src'
+import { Network } from '@helpers/network'
+import { AAVETokens, ONE } from '@oasisdex/oasis-actions/src'
 import BigNumber from 'bignumber.js'
 import { HardhatRuntimeEnvironment } from 'hardhat/types'
 
 import { swapUniswapTokens } from '../../helpers/swap/uniswap'
-import { mainnetAddresses } from '../../test/addresses/mainnet'
+import {
+  addressesByNetwork,
+  isMainnetByNetwork,
+  isOptimismByNetwork,
+  NetworkAddressesForTests,
+} from '../../test/test-utils/addresses'
 import { RuntimeConfig } from '../types/common'
 
 export type AAVETokensToGet = Exclude<AAVETokens, 'ETH' | 'WETH'>
-const tokens: Record<AAVETokensToGet, string> = {
-  STETH: mainnetAddresses.STETH,
-  WBTC: mainnetAddresses.WBTC,
-  USDC: mainnetAddresses.USDC,
-  WSTETH: mainnetAddresses.WSTETH,
-}
 
 export function buildGetTokenFunction(
   config: RuntimeConfig,
   hre: HardhatRuntimeEnvironment,
+  network: Network.MAINNET | Network.OPT_MAINNET,
+  wethAddress: string,
 ): (symbol: AAVETokensToGet, amount: BigNumber) => Promise<boolean> {
   return async function getTokens(symbol: AAVETokensToGet, amount: BigNumber): Promise<boolean> {
     /* Ensures we always have enough tokens to open a position */
     const BUFFER_FACTOR = 1.1
     const amountInInWeth = amount.times(BUFFER_FACTOR).toFixed(0)
+
+    let addresses: NetworkAddressesForTests | undefined
+    if (isMainnetByNetwork(network)) {
+      addresses = addressesByNetwork(Network.MAINNET)
+    }
+    if (isOptimismByNetwork(network)) {
+      addresses = addressesByNetwork(Network.OPT_MAINNET)
+    }
```

Comment:
```suggestion
    let addresses:  = addressesByNetwork(network)
```

So you are marking that the return type is of type `NetworkAddressesForTests`. Calling with a specific network type doesn't return any different value than calling the method with just `network`.